src/App.js

```diff
@@ -16,49 +20,85 @@ import DatastoreProvider from './context/DatastoreProvider';
 
 export default function AuthExample() {
   if (`${process.env.REACT_APP_AUTH}` === 'false') {
-    return returnValue(true);
+    localStorage.removeItem('seenWelcome', 'yes');
+  }
+
+  const initialShowWelcome = !localStorage.getItem('seenWelcome') || !localStorage.getItem('token');
+  const [showWelcome, setShowWelcome] = React.useState(initialShowWelcome);
+
+  if (`${process.env.REACT_APP_AUTH}` === 'false') {
+    return returnValue(true, showWelcome, setShowWelcome);
   }
 
   let accessToken = localStorage.getItem('token');
 
   if (accessToken) {
     const loggedIn = validateAccessToken(accessToken);
-    return returnValue(loggedIn);
+    if (!loggedIn) {
+      localStorage.removeItem('seenWelcome');
+    }
+    return returnValue(loggedIn, showWelcome, setShowWelcome);
   }
 
   const { hash } = window.location;
   const urlParams = new URLSearchParams(hash);
   accessToken = urlParams.get('access_token');
 
   if (accessToken) {
+    localStorage.setItem('seenWelcome', 'yes');
     localStorage.setItem('token', accessToken);
     window.history.replaceState({}, document.title, '/');
-    return returnValue(true);
+    return returnValue(true, showWelcome, setShowWelcome);
   }
 
-  return returnValue(false);
+  return returnValue(false, showWelcome, setShowWelcome);
 }
 
-const returnValue = (loggedIn) => (
-  <BrowserRouter>
-    <Switch>
-      <Route path="/auth">
-        <Auth />
-      </Route>
-      <Route exact path="/">
-        {loggedIn
-          ? (
-            <DatastoreProvider>
-              <Dashboard />
-            </DatastoreProvider>
-          )
-          : <Redirect to="/auth" />}
-      </Route>
-      <Route path="*">
-        <NotFound />
-      </Route>
-    </Switch>
-  </BrowserRouter>
+const action = (setShowWelcome) => (
+  <IconButton
+    size="small"
+    aria-label="close"
+    color="inherit"
+    onClick={() => setShowWelcome(false)}
+  >
+    <CloseIcon fontSize="small" />
+  </IconButton>
+);
+
+const returnValue = (loggedIn, showWelcome, setShowWelcome) => (
+  <>
+    <Snackbar
+      open={showWelcome}
+      autoHideDuration={6000}
+      anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+      onClose={() => setShowWelcome(false)}
+      message="Welcome to Saraswati, where knoweledge is power"
```

Comment:
Let's slap a period on the end of that and correct spelling to "knowledge"

Reply:
I almost prefer the irony

Reply:
👍